### PR TITLE
uri-scheme PR

### PR DESCRIPTION
**Pull request by paOol**
extracted only relevant changes in this new PR.

### CODE REVIEW

**Comment by SpicyPete** [comment]
Looking better/cleaner
Testing now, only a few odd flows which could use minor improvements before this is production ready I think

File: screens/SendSetupScreen.js [hunk 1]
```diff
@@ -295,7 +282,134 @@ const SendSetupScreen = ({
...
   };
 
+  const handleDeepLink = ({
+    address,
+    amount,
+    tokenAmount,
+    label,
+    tokenId,
+    symbol
+  }): DeepLinkParams => {
```
Comment (SpicyPete) [inline screens/SendSetupScreen.js]: The type `DeepLinkParams` is not defined anywhere, please create a type for it

File: screens/SendSetupScreen.js [hunk 2]
```diff
@@ -295,7 +282,134 @@ const SendSetupScreen = ({
...
+        );
+      } catch (error) {
+        setErrors(["invalid amount"]);
+      }
+    }
+
+    setToAddress(address);
+  };
+
+  const parseQr = (qrData: string, tokensById: object) => {
```
Comment (SpicyPete) [inline screens/SendSetupScreen.js]: change the type `object` to `any`, or enter it's actual type.

object with a lowercase `o` isn't a think, and `Object` with an uppercase `O` is being deprecated for `any`
https://flow.org/en/docs/types/objects/#toc-object-type

File: screens/SendSetupScreen.js [hunk 3]
```diff
@@ -341,16 +455,33 @@ const SendSetupScreen = ({
...
+                  } = parseQr(qrData, tokensById);
 
                   setErrors([]);
                   setQrOpen(false);
+
+                  // coinName and imageSource are consts, so redirect solves the problem
+                  // of incorrect names when in bch || slp.
+                  return navigation.navigate({
+                    routeName: "SendStack",
+                    key: Math.random() * 10000,
```
Comment (SpicyPete) [inline screens/SendSetupScreen.js]: What's the purpose of `key` in this navigation call?

Comment (paOol) [reply screens/SendSetupScreen.js]: needed to redirect onto the same stack.

a non random key might result in the screen not changing.

**Comment by SpicyPete** [comment]
Testing on real devices all flows

**Pull request merged**